gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -232,6 +235,43 @@ VectorValues HybridBayesNet::optimize(const DiscreteValues &assignment) const {
   return gbn.optimize();
 }
 
+/* ************************************************************************* */
+HybridValues HybridBayesNet::sample(VectorValues given, std::mt19937_64 *rng,
+                                    SharedDiagonal model) const {
```

Comment:
Interesting. I followed the signature in the other sample methods, so maybe update there as well?

Also it can't be `const` since DiscreteBayesNet does a `sampleInPlace(&given)` and `GaussianBayesNet` has the line `given.insert(sampled);`. Would you like me to update those to make them functional?

Reply:
Oh, now I remember! It is passed by value deliberately, to allow for updating it in-place.

Reply:
Should add that in comment so we don’t forget again

Reply:
I updated `GaussianBayesNet::sample` to be functional. So instead of `given.insert(sampled)`, it is now `result.insert(sampled)` with `result` initialized as `VectorValues result(given);`.
Perfectly backwards compatible. :)

Reply:
Yeah, not really :-) the argument was passed *by value* which makes a copy, functionally equivalent to what you changed it to. Modulo possible compiler optimizations that might do something different - no idea which one is best.